### PR TITLE
[FEAT]: Connected Explore and Bookmarks data to API

### PR DESCRIPTION
**Pull request by shivanigangadharan**
- Data on explore/bookmarks page connected to API
- Add/Remove Bookmarks

### CODE REVIEW

File: src/components/post/post.js [hunk 1]
```diff
@@ -1,28 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './post.css';
 import '../../styles.css';
 import avatar from '../../assets/defaultImg.png';
+import axios from 'axios';
+import { useAuth } from '../../context/authContext';
```
Comment (sharathnair9999) [inline src/components/post/post.js:6]: put all the context related part of code into an index.js file and import it from there to make it more readable

File: src/components/post/post.js [hunk 2]
```diff
@@ -1,28 +1,71 @@
...
+        try {
+            const res = await axios.post(`/api/users/remove-bookmark/${_id}`, {},
+                {
+                    headers: {
+                        authorization: encodedToken
+                    }
+                });
+            setUser({ ...user, bookmarks: res.data.bookmarks })
+        } catch (e) {
+            console.log(e);
```
Comment (sharathnair9999) [inline src/components/post/post.js:43]: you could show an alert for the error

**Review by sharathnair9999 — APPROVED** [review]
Everything looks good

File: src/context/authContext.js [hunk 3]
```diff
@@ -35,6 +35,7 @@ export const AuthProvider = ({ children }) => {
                 email: email,
                 password: password
             });
+            console.log(res);
```
Comment (prathmeshjagtap) [inline src/context/authContext.js]: you can remove console.log

**Review by prathmeshjagtap — APPROVED** [review]
Code is clean and well structured

File: src/context/authContext.js (hunk 3 above)
Comment (shivanigangadharan) [reply src/context/authContext.js]: done

File: src/components/post/post.js (hunk 2 above)
Comment (shivanigangadharan) [reply src/components/post/post.js:43]: done

File: src/components/post/post.js (hunk 1 above)
Comment (shivanigangadharan) [reply src/components/post/post.js:6]: understood, thanks for this suggestion will implement it.

**Pull request merged**